cfme/tests/openstack/cloud/test_volume_backup.py

```diff
@@ -64,6 +67,60 @@ def incremental_backup(volume_backup, provider):
         logger.warning('Exception during volume backup deletion - skipping..')
 
 
+@pytest.fixture(scope='function')
+def new_instance(provider):
+    prov_data = provider.data['provisioning']
+    prov_form_data = {
+        'request': {'email': fauxfactory.gen_email(),
+                    'first_name': fauxfactory.gen_alpha(),
+                    'last_name': fauxfactory.gen_alpha()},
+        'catalog': {'num_vms': '1',
+                    'vm_name': fauxfactory.gen_alpha()},
+        'environment': {'cloud_network': prov_data['cloud_network'],
+                        'cloud_tenant': prov_data['cloud_tenant']},
+        'properties': {'instance_type': partial_match(prov_data['instance_type'])},
+    }
+
+    instance_name = prov_form_data['catalog']['vm_name']
+
+    try:
+        instance = provider.appliance.collections.cloud_instances.create(
+            instance_name,
+            provider,
+            prov_form_data, find_in_cfme=True
+        )
+
+    except KeyError:
+        # some yaml value wasn't found
+        pytest.skip('Unable to find an image map in provider "{}" provisioning data: {}'
+                    .format(provider, prov_data))
+
+    yield instance
+
+    instance.cleanup_on_provider()
+
+
+@pytest.fixture(scope='function')
+def incremental_backup_of_attached_volume(appliance, provider, volume_backup, new_instance):
+    volume = appliance.collections.volumes.instantiate(volume_backup.volume, provider)
+    backup_collection = appliance.collections.volume_backups.filter({'provider': provider})
+    initial_volume_count = new_instance.volume_count
+    new_instance.attach_volume(volume.name)
+
+    wait_for(lambda: new_instance.volume_count > initial_volume_count,
+             delay=10,
+             timeout=300, message="Waiting for volume to be attached to instance",
+             fail_func=new_instance.refresh_relationships)
+
+    backup_name = fauxfactory.gen_alpha()
+    volume.create_backup(backup_name, incremental=True, force=True)
+    force_backup = backup_collection.instantiate(backup_name, provider)
+    yield force_backup
+
+    if force_backup.exists:
+        backup_collection.delete(force_backup)
+
```

Comment:
What about detaching the volume as well during cleanup?

Reply:
I am handling it on attached_volume fixture, after changes you requested